### PR TITLE
Milestone 2: Send and receive request to leaderboard API

### PR DESCRIPTION
**Pull request by DuaneDave**
### In this milestone, we:

- Made a request to a third-party API to get data.
- Sent a request to the same API to post the user's entry.
- Add a modal pop-up on validation to the input field to prevent empty entries.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1 +1,30 @@
...
+
+  if (!nameInput || !scoreInput) {
+    validate();
+  }
+
+  const data = {
+    user: nameInput.trim(),
+    score: scoreInput,
+  };
+
```
Comment (HAMMAS-SALEEM) [inline src/index.js:27]: license-link

- [ ] You have added a great readme but you didn't add a working link to your LICENSE file. Please make your link as given below :arrow_down:

`This project is licensed by [MIT](./LICENSE).`

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

### Hey @DuaneDave :wave:

### Highlights :1st_place_medal:
 
:heavy_check_mark: Meaningful commits
:heavy_check_mark: Added .gitignore
:heavy_check_mark: Following GitFlow
:heavy_check_mark: Using Async Await

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional Suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @HAMMAS-SALEEM in your question so I can receive the notification. Otherwise reach out to me on slack which is more fast. You can reach out to me on slack by searching my name in **full-time-students** channel :+1:

_**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_

------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (DuaneDave) [reply src/index.js:27]: Hello @HAMMAS-SALEEM!

Is this not supposed to have been an optional suggestion?
I really do not understand how this is a major blocker on why this PR shouldn't be merged pls I need further clarifications.

Comment (HAMMAS-SALEEM) [reply src/index.js:27]: @DuaneDave You should have a descriptive readme file. So this should not be in optional.
Please see the the requirement which says **Document your work in a professional way**
Screenshot from 2022-08-24 18-56-54 (copy)

Comment (DuaneDave) [reply src/index.js:27]: Okay

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @DuaneDave  

Your project is complete! Great job for making the requested changes🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- Validations correctly implemented
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**